src/Bitbucket.Cloud.Net/BitbucketCloudClient.cs

```diff
@@ -13,100 +14,110 @@
+            .WithAuthentication(_auth);
+
+        private async Task<TResult> ReadResponseContentAsync<TResult>(HttpResponseMessage responseMessage, Func<string, TResult> contentHandler = null)
+        {
+            string content = await responseMessage.Content.ReadAsStringAsync().ConfigureAwait(false);
+            return contentHandler != null
+                ? contentHandler(content)
+                : JsonConvert.DeserializeObject<TResult>(content);
+        }
+
+        private async Task<bool> ReadResponseContentAsync(HttpResponseMessage responseMessage)
+        {
+            string content = await responseMessage.Content.ReadAsStringAsync().ConfigureAwait(false);
+            return content.Length == 0;
+        }
+
+        private async Task HandleErrorsAsync(HttpResponseMessage response)
+        {
+            if (!response.IsSuccessStatusCode)
+            {
+                var errorResponse = await ReadResponseContentAsync<ErrorResponse>(response).ConfigureAwait(false);
+
+                string errorMessage = string.Join(Environment.NewLine, errorResponse?.Error?.Message);
+                throw new InvalidOperationException($"Http request failed ({(int)response.StatusCode} - {response.StatusCode}):\n{errorMessage}");
+            }
+        }
+
+        private async Task<TResult> HandleResponseAsync<TResult>(HttpResponseMessage responseMessage, Func<string, TResult> contentHandler = null)
+        {
+            await HandleErrorsAsync(responseMessage).ConfigureAwait(false);
+            return await ReadResponseContentAsync(responseMessage, contentHandler).ConfigureAwait(false);
+        }
+
+        private async Task<bool> HandleResponseAsync(HttpResponseMessage responseMessage)
+        {
+            await HandleErrorsAsync(responseMessage).ConfigureAwait(false);
+            return await ReadResponseContentAsync(responseMessage).ConfigureAwait(false);
+        }
+
+        private async Task<IEnumerable<T>> GetPagedResultsAsync<T>(int? maxPages, IDictionary<string, object> queryParamValues, Func<IDictionary<string, object>, Task<PagedResults<T>>> selector)
+        {
+            var results = new List<T>();
+            bool isLastPage = false;
+            int numPages = 0;
+
+            while (!isLastPage && (maxPages == null || numPages < maxPages))
+            {
+                var selectorResults = await selector(queryParamValues).ConfigureAwait(false);
+                selectorResults.Page = Math.Max(selectorResults.Page, 1);
+                results.AddRange(selectorResults.Values);
+
+                isLastPage = selectorResults.Next == null;
+
+                // parse out the next page id from the results
+                if (!isLastPage)
+                {
+                    var myUri = new Uri(selectorResults.Next);
+                    var nextPage = myUri.DecodeQueryParameters().FirstOrDefault(x => x.Key == "page");

```

Comment:
parse out the page id from the "next" url and use that for the next pge versus auto incrementing